### ISSUE TITLE
secure_validatable not working in Rails 5.2?

### ISSUE THREAD

**Issue by garyfoster** [devise-security/devise-security#65]
Anyone else notice this? Something funky is going on, I can't run rake tasks for example like db:create, I get an error on the devise line in the User model indicating that no db exists. If I remove `secure_validable` the error goes away. I can't yet tell if it's an issue with Rails 5.2 or an interaction with an other gem but thought I'd report it here in case someone else has run into it. I can provide more details later, but I'm tempted to just do the password validation myself and move on.

**Comment by olbrich** [comment]
@garyfoster did you sort out your issue or is there something we need to be looking into?

**Comment by garyfoster** [comment]
I didn't, but if I temporarily remove secure_validatable, run rake db:create then add it back, it gets around the issue. I was going to spend some more time trying to isolate where the exception is originating and add a new issue with better details. I'll reopen this for now just in case someone else needs to refer to it.

**Comment by garyfoster** [comment]
Ok, I haven't dug into this yet, but for starters, can you reproduce it? On a Rails 5.2 project with devise-security and the secure_validatable option in the devise line in User.rb.. Try to `rake db:drop` then `rake db:create` and do you get an error? If not, I can try to send you a minimal rails app that repro's.

**Comment by olbrich** [comment]
@garyfoster FYI, `secure_validateable` currently requires the use of `database_authenticatable`.  However the currently released code doesn't actually enforce that, so if you haven't done the database migrations required by that one, this could be the source of trouble.  Do you plan to use `database_authenticatable`?  I'll try to dig into a reproduction case, but it may take some time.

**Comment by garyfoster** [comment]
Yes, we use `database_authenticatable` on all our projects. 3 of our projects use devise-security, all of which have this same issue. It does have a fairly simple workaround but might be good to figure it out one day. I'll report back if we come up with anything. I don't think it's a migration issue. Something that happens in the environment is requiring that the db exists. When running `rake db:create`, of course the db doesn't exist yet, but that rake task is triggering something that is trying to access it.
